geopandas/geodataframe.py

```diff
@@ -898,15 +898,20 @@ def iterfeatures(self, na="null", show_bbox=False, drop_id=False):
         if len(properties_cols) > 0:
             # convert to object to get python scalars.
             properties = self[properties_cols].astype(object).values
+            na_mask = pd.isna(properties)
+
             if na == "null":
                 properties[pd.isnull(self[properties_cols]).values] = None
```

Comment:
```suggestion
                properties[na_mask] = None
```

I think we can reuse the mask here, to avoid calculating it twice?